File: backend-microservices/user-service/src/main/java/com/vanhuy/user_service/dto/UserDTO.java
package com.vanhuy.user_service.dto;

import lombok.Builder;
import lombok.Data;

import java.util.Set;

@Data
@Builder
public class UserDTO {
    private Integer userId;
    private String username;
    private String email;
    private String password;
    private String address;
    private String profileImageName;
    private Set<String> roles;
    private boolean isActive;
}

File: frontend/src/app/dto/auth/UserDTO.ts
export interface UserDTO {
    userId: number;
    username : string;
    email: string;
    address?: string;
    profileImageUrl?: string;
    roles: string[];
    active: boolean;
}

export interface UserCreateDTO extends Omit<UserDTO, 'userId'> {
    password: string;
 }

 export interface UserUpdateDTO extends Partial<UserCreateDTO> {
    
 }